File: nodeseteditor.client/src/model/WorkspaceDescription.ts
export const URN_PREFIX = 'urn:uuid:';

/** Extract the GUID from a urn:uuid: URI, or return the string as-is. */
export function urnToId(urn?: string): string {
   if (urn?.startsWith(URN_PREFIX)) return urn.substring(URN_PREFIX.length);
   return urn ?? '';
}

export function idToUrn(id: string): string {
   if (id.startsWith(URN_PREFIX)) return id;
   return URN_PREFIX + id;
}

export interface LocalizedText {
   locale?: string;
   text?: string;
}

export interface ApplicationDescription {
   applicationUri?: string;
   productUri?: string;
   applicationName?: LocalizedText;
   discoveryUrls?: string[];
   isDefault?: boolean;
}

export interface ServerConfiguration {
   endpointUrl?: string;
   transportProfileUri?: string;
   securityMode?: string;
   securityPolicyUri?: string;
}

export interface WorkspaceDescription extends ApplicationDescription {
   description?: LocalizedText;
   serverConfiguration?: ServerConfiguration;
   owner?: string;
   isOwner?: boolean;
   /** True if the current user may edit this workspace. Only the owner can
       write; shared (read-only) users and legacy no-owner workspaces are false. */
   canWrite?: boolean;
   acl?: string[];
   createdAt?: string;
   modifiedAt?: string;
}

export interface PaginatedResponse<T> {
   results: T[];
   totalCount?: number;
}

File: NodeSetEditor.Server/Model/WorkspaceDescription.cs
using System;
using System.Text.Json.Serialization;
using Opc.Ua.RestfulApi;

namespace NodeSetEditor.Server.Model
{
    public class WorkspaceDescription : ApplicationDescription
    {
        [JsonPropertyName("description")]
        public LocalizedText? Description { get; set; }

        [JsonPropertyName("serverConfiguration")]
        public ServerConfiguration? ServerConfiguration { get; set; }

        [JsonPropertyName("owner")]
        public string? Owner { get; set; }

        [JsonPropertyName("isOwner")]
        public bool? IsOwner { get; set; }

        /// <summary>
        /// True if the current user may modify this workspace (its content,
        /// models, metadata, and ACL). Only the owner can write; users the
        /// workspace is shared with are read-only.
        /// </summary>
        [JsonPropertyName("canWrite")]
        public bool? CanWrite { get; set; }

        [JsonPropertyName("acl")]
        public List<string>? Acl { get; set; }

        [JsonPropertyName("createdAt")]
        public DateTime? CreatedAt { get; set; }

        [JsonPropertyName("modifiedAt")]
        public DateTime? ModifiedAt { get; set; }
    }
}
